lib/gitsh/magic_variables.rb

```diff
@@ -4,9 +4,13 @@ def initialize(repo)
       @repo = repo
     end
 
-    def [](key)
+    def fetch(key)
       if available_variables.include?(key)
         send(key)
+      elsif block_given?
```

Comment:
It looks like this is only used from `Environment`, and a block is always provided.